pkg/controller/machine.go

```diff
@@ -503,7 +503,7 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
 			timeOut := metav1.Now().Add(-timeOutDuration).Sub(machine.Status.CurrentStatus.LastUpdateTime.Time)
 			timeOutOccurred = timeOut > 0
 
-			if forceDeleteLabelPresent || timeOutOccurred || lastDrainFailed {
+			if forceDeleteLabelPresent || timeOutOccurred {
 				// To perform forceful machine drain/delete either one of the below conditions must be satified
 				// 1. force-deletion: "True" label must be present
 				// 2. Deletion operation is more than drain-timeout minutes old
```

Comment:
The third point needs to be removed. The following comment in this `if` block also needs to be updated accordingly:
```
glog.V(2).Infof("Force deletion has been triggerred for machine %q due to Label:%t, timeout:%t, lastDrainFailed:%t",...
```

Also, we don't need the variable `lastDrainFailed` anymore. We can leave it for logging purposes though.

Reply:
thanks, good catch, fixed.